### PR TITLE
fix: Wrong padding on "Invite Team Members" modal (CALCOM-8414-FIX)

### LINKED ISSUE #9399: [CAL-1908] UI: move "Copy invite link" box out of Modal Footer
instead of: CleanShot 2023-06-07 at 18 52 12@2x

we should move it into the body of the modal to prevent text line break 

CleanShot 2023-06-07 at 18 53 11@2x

<sub>[CAL-1908](https://linear.app/calcom/issue/CAL-1908/ui-move-copy-invite-link-box-out-of-modal-footer)</sub>

### PR DESCRIPTION
**Pull request by gitstart-calcom**
## What does this PR do?

We are solving the spacing issues on "Invite Team Members" and "Add a Questions" modals

Invite Team Members (Invite Individual):
image

Invite Team Members (Bulk Import):
image

Add a Question:
image

### CODE REVIEW

**Comment by PeerRich** [comment]
i think the entire modal component was touched, not the team invite specifically

**Comment by gitstart-calcom** [comment]
@PeerRich we will take a look, thanks!

**Comment by PeerRich** [comment]
CleanShot 2023-06-01 at 22 11 12@2x

there still seems to be too much padding bottom here. can you check in with the figma and get the correct specifications?

**Comment by Jaibles** [comment]
> CleanShot 2023-06-01 at 22 11 12@2x
> 
> there still seems to be too much padding bottom here. can you check in with the figma and get the correct specifications?

CleanShot 2023-06-01 at 22 49 42@2x

https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/1.-Cal-DS?type=design&node-id=29898%3A100590&t=62LvZCZAEZm5zDdw-1

**Comment by gitstart-calcom** [comment]
Hey @PeerRich @Jaibles, should we use 16px or 24px on the padding bottom? Also, we think this prop "showDivider" can cause some bugs in other components in the future.

**Comment by PeerRich** [comment]
> Hey @PeerRich @Jaibles, should we use 16px or 24px on the padding bottom? Also, we think this prop "showDivider" introduces the problem, take a look [here](https://github.com/calcom/cal.com/commit/d63e7372cb56a66891f78d4269c6b11ebc8ea8f5#diff-4a6a4112d842250d02c624b09e247809c9da01832e009d08a04c9855f49f4e93R216). This prop can cause some bugs in other components in the future

as you see in the figma spec 16px

please figure out how to fix modals with showDivider as well

**Comment by PeerRich** [comment]
@gitstart-calcom 

we're getting closer. unfortunately now the normal modal has wrong padding:

CleanShot 2023-06-03 at 09 41 14@2x

CleanShot 2023-06-03 at 09 41 36@2x

**Comment by gitstart-calcom** [comment]
@PeerRich just to double confirm, the anatomy should be the same for both right?

in this case, the 40px + 16px between the last input and the button is missing, I believe it is not being respected in the other modal too, we are on it
let me know if it is about something else. 

image

**Comment by PeerRich** [comment]
yes both modals should have the same position of the buttons

**Comment by gitstart-calcom** [comment]
- On "MemberInvitationModal" the "showDivider" prop have unexpected behavior, so to fix this we moved the DialogFooter component outside the Form component to math how is being used in other places: 

image


- When the showDivider is not being used, we used the 40px distance instead 56px because it was looking too big and triggering the scroll, but this was a suggestion we can change quickly if you want to:

Before:
image


Now:
image

**Comment by Jaibles** [comment]
The **Add new team event type** modal should have a modal footer with a divider on it too. All creation modals should.

The only modals that do not have the divider are confirmation and destruction modals..
CleanShot 2023-06-07 at 08 00 12@2x


[View in Figma](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/1.-Cal-DS?type=design&node-id=29898%3A100590&t=WDWix2WUdFz5JrDr-1)

**Review by zomars — DISMISSED** [review]
Modal changes are breaking tests. Blocking until fixed 🙏🏽

File: packages/features/ee/teams/components/MemberInvitationModal.tsx [hunk 1]
```diff
@@ -271,24 +269,22 @@ export default function MemberInvitationModal(props: MemberInvitationModalProps)
...
-            <Button
-              type="submit"
-              color="primary"
-              className="ms-2 me-2"
-              data-testid="invite-new-member-button">
-              {t("send_invite")}
-            </Button>
-          </DialogFooter>
+          </div>
         </Form>
```
Comment (zomars) [inline packages/features/ee/teams/components/MemberInvitationModal.tsx:273]: This is the issue. Then submit buttons aren't inside the form anymore so the form never gets submitted.

Comment (zomars) [reply packages/features/ee/teams/components/MemberInvitationModal.tsx:273]: This is breaking functionality BTW.

Comment (zomars) [reply packages/features/ee/teams/components/MemberInvitationModal.tsx]: Pushed a fix. The form still looks fine.

image

Waiting for e2e tests to pass again before merging.

Comment (PeerRich) [reply packages/features/ee/teams/components/MemberInvitationModal.tsx]: yes perfect. i had merged a broken merge conflict

**Review by zomars — APPROVED** [review]

File: packages/features/ee/teams/components/MemberInvitationModal.tsx [hunk 2]
```diff
@@ -131,7 +131,7 @@ export default function MemberInvitationModal(props: MemberInvitationModalProps)
         </div>
 
         <Form form={newMemberFormMethods} handleSubmit={(values) => props.onSubmit(values)}>
-          <div className="space-y-6">
+          <div className="my-6 space-y-6">
```
[the remark this answers is not in the document]
Comment (juliopiubello) [reply packages/features/ee/teams/components/MemberInvitationModal.tsx:134]: @zomars 

image

[the remark this answers is not in the document]
Comment (gitstart-calcom) [reply packages/features/ee/teams/components/MemberInvitationModal.tsx:134]: @zomars 
image

**Comment by PeerRich** [comment]
I think i fixed it now for both dialogs with and without divider

**Review by PeerRich — APPROVED** [review]
great stuff. done with all small fixes

**Pull request merged**